src/row-builder.js

```diff
@@ -150,17 +156,43 @@ RowBuilder.prototype.build = function() {
 
     previousChunk = chunk;
   });
+
+  // As chunks are now in rows, remove them.
+  this.chunks.length = 0;
+
+  // Examine the last chunk. If it was 'chunked', strip that off and place it
+  // back to be processed in the next build.
+  if (previousChunk && previousChunk.chunkedValue) {
+    previousChunk.values = previousChunk.values.splice(-1);
+    this.chunks = [previousChunk];
+    // There is a partial row added to rows. Remove this.
+    this.currentRow.pop();
+  }
 };
 
 /**
- * Transforms values into JSON format.
+ * Flush already complete rows.
  */
-RowBuilder.prototype.toJSON = function() {
-  this.build();
+RowBuilder.prototype.flush = function() {
+  var rowsToReturn = this.rows;
 
+  if (!is.empty(this.rows[0]) && this.currentRow.length !== this.fields.length) {
```

Comment:
@crwilcox While the previous changes I pointed out were trying to preserve old behavior, this is 1/2 changes to the logic that you wrote:

```diff
-if (!is.empty(this.rows)
+if (!is.empty(this.rows[0])
```

This is because `this.rows` should never be empty, as I understand it. When we reset `this.rows`, we set it to: `[[]]`. So `this.rows` will always have an object, and the is-empty check will always be false.